src/App.js

```diff
@@ -1,40 +1,78 @@
-import React, { useState, useEffect } from 'react';
-import { SearchBar } from './modules/recipes/search-bar';
-import { fetchRecipesByIngredient, fetchRecipeDetailsById } from './modules/recipes/recipeService';
-import './App.css';
-import { RecipeList } from './modules/recipes/recipe-list/RecipeList';
-import { RecipeDetail } from './modules/recipes/recipe-detail/RecipeDetail.js'; 
+import { useState } from "react";
+import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
+import { UserProvider } from "./Contexts/UserContext";
+import { NavBar } from "./shared-components/NavBar";
+import { SearchBar } from "./modules/recipes/SearchBar";
+import { RecipeList } from "./modules/recipes/RecipeList";
+import { RecipeDetails } from "./modules/recipes/RecipeDetails";
+import { Categories } from "./modules/recipes/Categories";
+import { CategoryDetails } from "./modules/recipes/CategoryDetails";
+import { CreateRecipe } from "./modules/recipes/CreateRecipe";
+import { ProtectedRoute } from "./ProtectedRoute";
+import { fetchRecipeDetailsById } from "./modules/recipes/recipeService";
+import { useEffect } from 'react';
+import "./App.css";
 
-export const App = () => {
-    const [recipes, setRecipes] = useState([]);
-    const [selectedRecipe, setSelectedRecipe] = useState(null); 
+export const App = ({ recipes, onRecipeClick }) => {
+  const [selectedRecipe, setSelectedRecipe] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
First error: you define the initial value as true. The value is never updated after so that the loading indicator is always shown on the screen. 
Remove this state completely, you don't need to load any data in the App component right now.